src/runtime/composables/useIsSanityLivePreview.ts

```diff
@@ -0,0 +1,18 @@
+import { useSanityPreviewEnvironment } from './useSanityPreviewEnvironment'
+
+/**
+ * Detects if the application is considered to be in a "Live Preview" mode. Live
+ * Preview means that the application is either being viewed inside Sanity
+ * Presentation Tool or being previewed in a standalone context. Useful for
+ * conditionally rendering UI that should only be visible to content editors.
+ * The composable returns `null` initially, to signal it doesn't yet know if
+ * it's live previewing or not. Then `true` if it is, and `false` otherwise.
+ * @public
+ */
+export function useIsSanityLivePreview(): Ref<boolean | null> {
+  return computed(() => {
+    const env = useSanityPreviewEnvironment().value
+    if (env === 'checking') return null
+    return ['presentation-iframe', 'presentation-window', 'live'].includes(env)
+  })
```

Comment:
we should probably move the composable calls out of the computed

```suggestion
  const env = useSanityPreviewEnvironment()
  return computed(() => {
    if (env.value === 'checking') return null
    return ['presentation-iframe', 'presentation-window', 'live'].includes(env.value)
  })
```

Reply:
Woops. Thank you!